build/ci/unit-tests-template.yml

```diff
@@ -4,6 +4,16 @@ jobs:
   pool: ${{ parameters.pool }}
   timeoutInMinutes: 20
   steps:
+    - checkout: none
+
+    - script: |
+        @echo on
+        @git version
+        git init
+        git fetch --no-auto-gc --no-recurse-submodules --no-show-forced-updates --no-tags --depth=1 "$(Build.Repository.Uri)" "$(Build.SourceVersion)"
+        git -c advice.detachedHead=false checkout --no-progress "$(Build.SourceVersion)"
+      displayName: Shallow Checkout
```

Comment:
If we're gonna use a script, might make sense to put it as an actual script file in this folder so it can be used by other yml pipelines too. You might need to pass in those 2 values as parameters. Then, you can also have some comments at the top (you can just copy/paste your comment in this PR about what the script does).